### DISCUSSION TITLE
Events and RSVPs

### DISCUSSION THREAD

**Discussion by ngerakines** [lexicon-community/lexicon#6] [Projects]
I want to create a new NSID and series of types to support the generic event and RSVP types within this lexicon group. These types would be part of the `community.lexicon.calendar`. I think "calendar" is a good word to represent grouping of events.

## Intent and Rationale

Events and the representation of someone's interest and intent to participate in the event (RSVP) are somewhat generic types that can be shared across applications, systems, and tools.

For example, the Smoke Signal lexicon was created using the base `events.smokesignal` NSID with event, rsvp, and location types, but they are not specific to Smoke Signal and can be used outside of that application.

### community.lexicon.calendar.event

The event type represents a general "event" that can be in-person, online, or hybrid.

```json
{
    "lexicon": 1,
    "id": "community.lexicon.calendar.event",
    "defs": {
        "main": {
            "type": "record",
            "description": "Record containing a calendar event.",
            "key": "tid",
            "record": {
                "type": "object",
                "required": [
                    "name",
                    "text",
                    "createdAt"
                ],
                "properties": {
                    "name": {
                        "type": "string",
                        "maxLength": 100,
                        "description": "The name of the event."
                    },
                    "text": {
                        "type": "string",
                        "maxLength": 3000,
                        "description": "The event description and content for the event."
                    },
                    "createdAt": {
                        "type": "string",
                        "format": "datetime",
                        "description": "Client-declared timestamp when this event was originally created."
                    },
                    "startsAt": {
                        "type": "string",
                        "format": "datetime",
                        "description": "Client-declared timestamp when this event starts."
                    },
                    "endsAt": {
                        "type": "string",
                        "format": "datetime",
                        "description": "Client-declared timestamp when this event ends."
                    },
                    "mode": {
                        "type": "ref",
                        "ref": "#mode",
                        "description": "The attendance mode of the event."
                    },
                    "status": {
                        "type": "ref",
                        "ref": "#status",
                        "description": "The status of the event."
                    },
                    "locations": {
                        "type": "ref",
                        "ref": "community.lexicon.calendar.location#locations",
                        "description": "The locations where the event takes place."
                    }
                }
            }
        },
        "mode": {
            "type": "string",
            "description": "The mode of the event.",
            "default": "community.lexicon.calendar.event#inperson",
            "knownValues": [
                "community.lexicon.calendar.event#virtual",
                "community.lexicon.calendar.event#inperson",
                "community.lexicon.calendar.event#hybrid"
            ]
        },
        "virtual": {
            "type": "token",
            "description": "A virtual event that takes place online."
        },
        "inperson": {
            "type": "token",
            "description": "An in-person event that takes place offline."
        },
        "hybrid": {
            "type": "token",
            "description": "A hybrid event that takes place both online and offline."
        },
        "status": {
            "type": "string",
            "description": "The status of the event.",
            "default": "community.lexicon.calendar.event#scheduled",
            "knownValues": [
                "community.lexicon.calendar.event#scheduled",
                "community.lexicon.calendar.event#rescheduled",
                "community.lexicon.calendar.event#cancelled",
                "community.lexicon.calendar.event#postponed"
            ]
        },
        "scheduled": {
            "type": "token",
            "description": "The event has been created and scheduled."
        },
        "rescheduled": {
            "type": "token",
            "description": "The event has been rescheduled."
        },
        "cancelled": {
            "type": "token",
            "description": "The event has been cancelled."
        },
        "postponed": {
            "type": "token",
            "description": "The event has been postponed and a new start date has not been set."
        }
    }
}
```

### community.lexicon.calendar.rsvp

The RSVP type represents a handle's interest in attending an event.

```json
{
    "lexicon": 1,
    "id": "community.lexicon.calendar.rsvp",
    "defs": {
        "main": {
            "type": "record",
            "description": "Record declaring an RSVP for an event.",
            "key": "tid",
            "record": {
                "type": "object",
                "required": [
                    "subject",
                    "status"
                ],
                "properties": {
                    "subject": {
                        "type": "ref",
                        "ref": "com.atproto.repo.strongRef"
                    },
                    "status": {
                        "type": "string",
                        "default": "community.lexicon.calendar.rsvp#interested",
                        "knownValues": [
                            "community.lexicon.calendar.rsvp#interested",
                            "community.lexicon.calendar.rsvp#going",
                            "community.lexicon.calendar.rsvp#notgoing"
                        ]
                    }
                }
            }
        },
        "interested": {
            "type": "token",
            "description": "Interested in the event"
        },
        "going": {
            "type": "token",
            "description": "Going to the event"
        },
        "notgoing": {
            "type": "token",
            "description": "Not going to the event"
        }
    }
}
```

### community.lexicon.calendar.location

The location types represents collections of and individual physical and virtual locations in which an event takes place.

```json
{
    "lexicon": 1,
    "id": "community.lexicon.calendar.location",
    "defs": {
        "locations": {
            "type": "array",
            "items": {
                "type": "ref",
                "ref": "community.lexicon.calendar.location#location"
            }
        },
        "location": {
            "type": "union",
            "refs": [
                "community.lexicon.calendar.location#place",
                "community.lexicon.calendar.location#virtual",
                "community.lexicon.calendar.location#h3geo",
            ]
        },
        "place": {
            "type": "object",
            "description": "A physical location in the form of an address.",
            "required": [
                "country"
            ],
            "properties": {
                "country": {
                    "type": "string",
                    "description": "The ISO 3166 country code. Preferably the 2-letter code.",
                    "minLength": 2,
                    "maxLength": 10
                },
                "postalCode": {
                    "type": "string",
                    "description": "The postal code of the location.",
                    "maxLength": 20
                },
                "region": {
                    "type": "string",
                    "description": "The administrative region of the country. For example, a state in the US.",
                    "maxLength": 100
                },
                "locality": {
                    "type": "string",
                    "description": "The locality of the region. For example, a city in the US.",
                    "maxLength": 100
                },
                "street": {
                    "type": "string",
                    "description": "The street address.",
                    "maxLength": 100
                },
                "name": {
                    "type": "string",
                    "description": "The name of the place.",
                    "maxLength": 100
                }
            }
        },
        "virtual": {
            "type": "object",
            "description": "A virtual location in the form of a uri.",
            "required": [
                "url"
            ],
            "properties": {
                "url": {
                    "type": "string",
                    "format": "uri",
                    "description": "The URL of the virtual location."
                },
                "name": {
                    "type": "string",
                    "description": "The title of the URL."
                }
            }
        },
        "h3geo": {
            "type": "object",
            "description": "A physical location in the form of a h3 encoded location",
            "required": [
                "value"
            ],
            "properties": {
                "value": {
                    "type": "string",
                    "description": "The h3 encoded location."
                },
                "name": {
                    "type": "string",
                    "description": "The name of the location."
                }
            }
        }
    }
}
```

**Comment by ngerakines** [comment]
I think one major area that I'd like feedback on is the concept of location collections and types of locations. In Smoke Signal, they've existed within the calendar namespace because they were used directly with events. I think this could be a candidate for moving out of that event-namespace level.

### community.lexicon.locations.place

```json
{
    "lexicon": 1,
    "id": "community.lexicon.locations.place",
    "defs": {
        "main": {
            "type": "object",
            "description": "A physical location in the form of an address.",
            "required": [
                "country"
            ],
            "properties": {
                "country": {
                    "type": "string",
                    "description": "The ISO 3166 country code. Preferably the 2-letter code.",
                    "minLength": 2,
                    "maxLength": 10
                },
                "postalCode": {
                    "type": "string",
                    "description": "The postal code of the location.",
                    "maxLength": 20
                },
                "region": {
                    "type": "string",
                    "description": "The administrative region of the country. For example, a state in the US.",
                    "maxLength": 100
                },
                "locality": {
                    "type": "string",
                    "description": "The locality of the region. For example, a city in the US.",
                    "maxLength": 100
                },
                "street": {
                    "type": "string",
                    "description": "The street address.",
                    "maxLength": 100
                },
                "name": {
                    "type": "string",
                    "description": "The name of the place.",
                    "maxLength": 100
                }
            }
        }
    }
}
```

### community.lexicon.locations.virtual

```json
{
    "lexicon": 1,
    "id": "community.lexicon.locations.virtual",
    "defs": {
        "main": {
            "type": "object",
            "description": "A virtual location in the form of a uri.",
            "required": [
                "url"
            ],
            "properties": {
                "url": {
                    "type": "string",
                    "format": "uri",
                    "description": "The URL of the virtual location."
                },
                "name": {
                    "type": "string",
                    "description": "The title of the URL."
                }
            }
        }
    }
}
```

### community.lexicon.locations.h3geo

```json
{
    "lexicon": 1,
    "id": "community.lexicon.locations.virtual",
    "defs": {
        "main": {
            "type": "object",
            "description": "A physical location in the form of a h3 encoded location",
            "required": [
                "value"
            ],
            "properties": {
                "value": {
                    "type": "string",
                    "description": "The h3 encoded location."
                },
                "name": {
                    "type": "string",
                    "description": "The name of the location."
                }
            }
        }
    }
}
```

**Comment by jay-elizaga-dev** [comment]
Looks great, I'm curious as to the design decision to not require a "name".  

To the h3 encoded location, I would add `H3 geospatial indexing system encoded location`, as many including me may not know what an h3 location is right away.  

For virtual location, is there a way to require based on an "or"?  Not all virtual locations may have a URL/URI

            "required": [
                "url" or "name"
            ],

**Reply by ngerakines** [reply]
Unfortunately there aren't conditional required fields. That isn't possible at this time.

**Comment by essential-randomness** [comment]
Sorry for being late to this, I see there's already a PR and I'll go add some further comments there but I'm putting the full feedback from our community discussion here for easier tracking.

## `community.lexicon.calendar.event`
### Additions

- **A top-level URL for further information about the event.** Most events will have their own official page, especially given that the text field doesn't have a way to embed images (I assume), and there may be other limitations. Yes, you could put it in the description, but that's going to make it a mess to have compact UIs that surface events and where to go to learn more (and many people do forget to put these links in descriptions fairly often).
- **An image for the event.**
- **Last updated date.** There was some discussions here about what's supposed to happen in AtProto when an event is modified, but in general it'd be nice to be able to tell when the event was last updated.
- **Rescheduled events should have a pointer to new event.** Alternative, there should be a pointer from an event to what event this is a rescheduling of. 

### Potential Changes

- **Text really needs a better name.** I saw (I think in the PR) something about it being there for compatibility with Bluesky's posts, but I couldn't find any information about where this type of compatibility is encouraged, and if this were the case I don't think it should be the event description, as much as "the event announcement" that will be shown in social updates, which can be two very separate things. Personally, I think any field with unclear semantic is going to be a huge headache in the long run, so generic names like "text" trigger alarm bells if it's meant to represent "description".
- **Text should also be optional.** Not all events need descriptions. Most of our community events actually don't have a description, because the context for most already fits in the title. Again, if this is about "what would be displayed in the socials announcement" I'd like to learn more about where this is explained/encouraged as a good practice.
- I saw someone already commenting about recurring events in the PR, but a +1 from us :) Maybe not MVP, but definitely something that should be considered soon.


Making different comments for the other types.

**Reply by ngerakines** [reply]
> A top-level URL for further information about the event. Most events will have their own official page, especially given that the text field doesn't have a way to embed images (I assume), and there may be other limitations. Yes, you could put it in the description, but that's going to make it a mess to have compact UIs that surface events and where to go to learn more (and many people do forget to put these links in descriptions fairly often).

Yes, I think a reasonable next step would be to introduce something like this to the event structure:

```json
{
  "uris": [
    {
      "uri": "https://example.com/"
    },
    {
      "name": "tickets",
      "uri": "https://example.com/"
    }
  ]
}
```

> An image for the event.

Images, attachments, and blob handling is a future thing. This is definitely important, and I want to think through the difference scenarios and use cases. It was intentionally left out in revision 1 because it adds a lot of complexity. 

> Last updated date.

Yeah, this is good idea. I'm really interested in discussing the idea of "editing" records more and getting some community feedback and consensus.

> Rescheduled events should have a pointer to new event. Alternative, there should be a pointer from an event to what event this is a rescheduling of.

This is another thing that I'd like to discuss further: tombstones

If a record is deleted (soft deleted) or is a pointer / link to a different record, what is does the pattern of an HTTP redirect look like in AT Protocol?

**Reply by ngerakines** [reply]
> Text really needs a better name. I saw (I think in the PR) something about it being there for compatibility with Bluesky's posts, but I couldn't find any information about where this type of compatibility is encouraged, and if this were the case I don't think it should be the event description, as much as "the event announcement" that will be shown in social updates, which can be two very separate things. Personally, I think any field with unclear semantic is going to be a huge headache in the long run, so generic names like "text" trigger alarm bells if it's meant to represent "description".

Yes, originally there was a `description` field but I renamed it to `text` when I also rename the `createdAt` field to bring it closer to the concept of a "text-centric record" in the ATmosphere. I'm not really against using `description` instead of `text`, but I have two thoughts:

1. Naming that field `text` is really self-descriptive in both what it is (it's the text / content of the record) and what format it's in (plain text).
2. Paul and the bsky team have commented that they'd like to change the "standard" text formatting system for feed posts and for compatibility, I'd like to implement and use whatever that is.

**Reply by ngerakines** [reply]
> Text should also be optional. Not all events need descriptions. Most of our community events actually don't have a description, because the context for most already fits in the title. Again, if this is about "what would be displayed in the socials announcement" I'd like to learn more about where this is explained/encouraged as a good practice.

Yeah, I'm in agreement here.

**Comment by essential-randomness** [comment]
I'll do the feedback on rsvp + geo stuff in one since the first one is small.

## `community.lexicon.calendar.rsvp`

Main feedback here is that "interested" is pretty vague in that we aren't sure if it semantically conveys a *maybe* (I want to come but might have a conflict) or a *keep me updated* (I want to stay updated about this, but I haven't committed to anything). These are very different things for both organizers and attendees. Personally, we'd like to see both:

- `community.lexicon.calendar.rsvp#interested`
- `community.lexicon.calendar.rsvp#maybe`

## `community.lexicon.calendar.location`

- **This is the one we feel most strongly about: "`#virtual`" should be renamed to "`#url`"**, as there are many virtual events that won't have an URL associated with it. Simple examples are events in private chats, but also events in non-public Discord servers, or servers whose URL is only known to attendees.
- There should be potentially "`#other`" type of location with just a name and maybe a optional description. Many events are community events where the community would have places they already know and that again don't have URLs associated. This could also be a way to eventually figure out what type of locations we aren't thinking about. Also it was pointed out some physical events don't have addresses, like those within cruise ships (see next point).
- Why is country the only required field in "place"? It feels very arbitrary. Also, some events that have a physical place might just have a name, like "[Name]'s house" if they're being organized by people who are all in the same circle.
- Unsure where the 100 characters limits is coming from in street, but it was pointed out that some reservations have pretty long addresses because they're more like "descriptions of directions". If this was taken from some existing standard this might be moot, but it could be worth doing some research or increasing the limit. We risk of running into a case like [Falsehoods Programmers Believe About Names](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/).

**Reply by jay-elizaga-dev** [reply]
> Main feedback here is that "interested" is pretty vague in that we aren't sure if it semantically conveys a maybe (I want to come but might have a conflict) or a keep me updated (I want to stay updated about this, but I haven't committed to anything).

> community.lexicon.calendar.rsvp#interested

Based on this, I would suggest `community.lexicon.calendar.rsvp#keepmeupdated` to reduce ambiguity.

**Reply by ngerakines** [reply]
Ok, I actually feel strongly about this, but I'm open to finding a good compromise.

"Going" and "Not Going" are pretty obvious. "Yes, I'm going to the event" or "No, I'm not going to the event" are given responses. What I was trying to do with "Interested" is find a functional third option to represent all of the following valid scenarios:

* I don't know if I can go or not yet.
* I want to acknowledge that I've seen the event and made my interest known, but also want to remain non-committal.

I thought "Interested" is a better alternative to "Maybe". Perhaps "Maybe" is more clear and direct in conveying non-commitment in the context of RSVP'ing to an event. I'm not sure I agree that both are needed, however.

I feel very strongly against making it a free-form string because I believe that both the user experience and integrator experience will be significantly compromised.

**Reply by ngerakines** [reply]
> This is the one we feel most strongly about: "#virtual" should be renamed to "#url", as there are many virtual events that won't have an URL associated with it. Simple examples are events in private chats, but also events in non-public Discord servers, or servers whose URL is only known to attendees.

This is language and terminology that I took from https://schema.org/VirtualLocation.

I'm not against changing it to url because that very plainly represents the type of content the record is.

**Reply by ngerakines** [reply]
> There should be potentially "#other" type of location with just a name and maybe a optional description. Many events are community events where the community would have places they already know and that again don't have URLs associated. This could also be a way to eventually figure out what type of locations we aren't thinking about. Also it was pointed out some physical events don't have addresses, like those within cruise ships (see next point).

> Why is country the only required field in "place"? It feels very arbitrary. Also, some events that have a physical place might just have a name, like "[Name]'s house" if they're being organized by people who are all in the same circle.

The location enum is open, so additional types can be added. I don't think adding an `#other` type is necessary now. Instead, I'd suggest integrators either create PR's to lexicon community or just reference their own lexicon for locations that aren't listed here.

The primary use case for a place is a physical place that someone can actually go to. There is no universal standard address and countries treat them differently, but that formatting is almost always country specific. So in order to give users and integrators the minimum amount of information to parse an address and understand how it should be captured and formatted, a country is required.

The bare minimum amount of information to represent a place is a name and a country. For example, if someone created an event with the place "The Grand Canyon, USA" then trying to cover a single administrative area or region could be tricky because spans 7 (or so?) of them.

**Reply by ngerakines** [reply]
> Unsure where the 100 characters limits is coming from in street, but it was pointed out that some reservations have pretty long addresses because they're more like "descriptions of directions". If this was taken from some existing standard this might be moot, but it could be worth doing some research or increasing the limit. We risk of running into a case like [Falsehoods Programmers Believe About Names](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/).

NGL this was totally arbitrary. I think it should just be removed.

**Reply by ebwinters** [reply]
`interested` and `maybe` serve 2 different purposes, it might be good to have both.
For example, here's a Facebook event for a free day at an art museum. They don't really care if I'm a maybe, the event is happening regardless of headcount. I'm responding yes/no to show my friends I'm going, and interested to follow event updates or show general interest in the event happenings
![image](https://github.com/user-attachments/assets/a6db74ba-f4d7-4c54-bee0-65f223bb3ce8)

Here is a Partiful invite. This event could be 5 people, and my being a `maybe` actually matters. I'm probably already interested in the event if I'm 1/5 people on the invite list
![image](https://github.com/user-attachments/assets/65c44bff-481e-4a7d-9968-378db3f100e5)

**Reply by Narkoleptika** [reply]
> Ok, I actually feel strongly about this, but I'm open to finding a good compromise.
> 
> "Going" and "Not Going" are pretty obvious. "Yes, I'm going to the event" or "No, I'm not going to the event" are given responses. What I was trying to do with "Interested" is find a functional third option to represent all of the following valid scenarios:
> 
>     * I don't know if I can go or not yet.
> 
>     * I want to acknowledge that I've seen the event and made my interest known, but also want to remain non-committal.
> 
> 
> I thought "Interested" is a better alternative to "Maybe". Perhaps "Maybe" is more clear and direct in conveying non-commitment in the context of RSVP'ing to an event. I'm not sure I agree that both are needed, however.
> 
> I feel very strongly against making it a free-form string because I believe that both the user experience and integrator experience will be significantly compromised.

Does the value need to be a string? It seems the main piece of information to capture here is commitment which could be represented as a percentage. That would allow implementations to use whatever labels with as much granularity as they like. eg:
```typescript
// TODO: Replace buttons with fancy commitment slider
const rsvpButtons = [
    {
        label: 'Yes',
        commitment: 1,
    },
    {
        label: 'Maybe',
        commitment: 0.6,
    },
    {
        label: 'Interested',
        commitment: 0.3,
    },
    {
        label: 'No',
        commitment: 0,
    },
]
```

**Reply by seitzme** [reply]
To me, "interested" implies a greater level of interest, and more strongly justifies sending informational updates. "Maybe" to me conveys a lower level of interest, more passive and more like a bookmark.

**Reply by thomasgwatson** [reply]
On the Yes, maybe, interested, keep-me-posted, no aspect of RSVP.

I'm not familiar with the schema rules for lexicons, so forgive me if this is non-compliant:

Is there a way to offer a broader set of valid responses (say, five options), and then enforce that an event record defines a subset of responses acceptable to that event? Whether that is [yes,no], [yes, maybe, no] or [ yes, interested, not interested]. This allows app builders to pick their preferred set of responses.

As we touch on the meaning/weight of various options (thinking of 'interested' vs 'maybe'), there is also the muddling factor of i18n; these expressions might have different weights, even to people speaking the same language, let alone when they are translated across languages. Perhaps that leans us towards including a `commitment` number.

**Reply by ngerakines** [reply]
I'm not going to move forward with making this a complex data structure. I believe pretty strongly that it'll introduce too much complexity and will lead to a bad user experience, misuse, and bugs in implementation.

*Happy Path*
1. User creates an RSVP of "Going" if they are going.

*Changed My Mind*
1. User creates an RSVP of "Going" if they are going.
2. User changes their existing RSVP to "Not Going" if they are no longer going.

*Will Not Go"
1. User creates an RSVP of "Not Going" if they want to explicitly note that they are not going.

*Non-Committed*
1. User creates an RSVP of "Interested"

**Comment by bmann** [comment]
I think interested is good! Depending on the application, I can see someone “favouriting” or “remind me later” using this field. 

You might be interested because you want to share it, decide later, or just literally make interesting events.

**Comment by pdelfan** [comment]
I agree with @ebwinters on the distinction; it mainly depends on the intended signal to the three groups who are viewing this decision: the user, the organizer, and other viewers. If I had to pick between the two, I'd go with `maybe` because it implies both interest and the possibility of attendance. 

I think `interested` has a higher chance of being misunderstood and used for unintended purposes. Maybe it can be a separate field that's used for saving events or adding it to your history. In that case, `save` can also be the option—there's probably a more intuitive option out there.

**Comment by thomasgwatson** [comment]
At the risk of prompting wider abstraction;

We have the concept of events

And a concept of an RSVP

But is there a concept for an invitation?

Invites allow event runners to more meaningfully track who they want at an event. 
- It gives them denominator on attendance; I invited 100 and have received 30 responses
- It would allow events to be closed to only those invited
- Event organizers would be able to understand who's invites were accepted (Jay's invites were all accepted but so far none of Alex's have been).

Thoughts?

**Reply by ngerakines** [reply]
There isn't, but I think that it's a really good idea and something that should be explored in the future.

**Comment by thomasgwatson** [comment]
hosts: should an event have a property that is an array of account ids, to represent the hosts of the event?

I don't know the full extend of 'ownership' of a record in ATprotocol.... can multiple entities be permitted to edit a record? Maybe only if they share a PDS?

**Reply by ngerakines** [reply]
I think we should punt this for now. At some point *in the future*:tm: we may have permissions and multi-did records, but until then we can somewhat avoid that problem by using the field `contacts` to represent who can be contacted regarding the event.

**Comment by mikestaub** [comment]
We could group the status on two axes: 'ability to attend' in x and 'desire to attend' in y. This mapping will give far more signal to the hosts which can be useful for planning and feeding the recommendation algorithms. 

(1,1)=going, (-1,1)=wish-i-could, (0,1)=will-try, (-1,-1)=cant-attend, (1, -1)=not-interested

**Reply by jay-elizaga-dev** [reply]
> We could group the status on two axes: 'ability to attend' in x and 'desire to attend' in y. This mapping will give far more signal to the hosts which can be useful for planning and feeding the recommendation algorithms.
> (1,1)=going, (-1,1)=wish-i-could, (0,1)=will-try, (-1,-1)=cant-attend, (1, -1)=not-interested

Representing two variables in different axes usually means they are orthogonal. 

Desire to attend and ability to attend seems to be related (non-orthogonal) by their variations of attendance.  

It would prob be better to use enumerations to denote the different states or levels of attendance:

Enums for Ability to attend: 
- Can Attend
- Cannot Attend
- Won't attend
- Will Attend
- ...

Enums for Desire to attend: 
- Going
- Wish I could
- Not interested
- etc

**Reply by jay-elizaga-dev** [reply]
Furthermore, the enumeration states can be mapped to a number based on their respective signals:
Enums for Ability to attend:
- Will Attend (1)
- Can Attend (0.5)
- ...
- Cannot Attend (-0.5)
- Won't attend (-1)
- ...

**Comment by ngerakines** [comment]
I'd like to thank everyone that has contributed to this initial discussion for calendar events and RSVPs.  I'm pretty happy with how https://github.com/lexicon-community/lexicon/pull/23 has turned out and I'll be merging it today.

I'm going to close this discussion to encourage new discussions be created for some of the additions and changes that were brought up here. Cheers!
